contracts/FragToken.sol

```diff
@@ -106,25 +106,26 @@ contract FRAGToken is ERC20, ERC20Permit, Ownable {
         require(amount > 0, "Amount must be greater than 0");
 
         // make sure the signature is valid
-        bytes32 hash = ECDSA.toEthSignedMessageHash(
+         bytes32 digest = _hashTypedDataV4(
             keccak256(
-                abi.encodePacked(
-                    "FragUnlock",
+                abi.encode(
+                    keccak256(abi.encodePacked("Msg(string name,address sender,uint256 amount)")),
+                    keccak256(abi.encodePacked("FragUnlock")),
                     msg.sender,
-                    uint64(block.chainid),
                     amount
                 )
             )
         );
+        
         require(
-            msg.sender == ECDSA.recover(hash, signature),
+            msg.sender == ECDSA.recover(digest, signature),
             "Invalid signature"
         );
 
         // reset the stake
-        _locksAmount[msg.sender] = 0;
-        _locksBlock[msg.sender] = 0;
-        _locktime[msg.sender] = 0;
+        delete _locksAmount[msg.sender];
```

Comment:
isn't it the same of 0 assign? curious.

Reply:
It's the same yes. I just found it cleaner.